projects/sbb-esta/angular-public/schematics/ng-add/setup-project.ts

```diff
@@ -0,0 +1,137 @@
+import { chain, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import {
+  addModuleImportToRootModule,
+  getAppModulePath,
+  getProjectFromWorkspace,
+  getProjectMainFile,
+  hasNgModuleImport
+} from '@angular/cdk/schematics';
+import { getWorkspace, updateWorkspace } from '@schematics/angular/utility/config';
+
+import { Schema } from './schema';
+
+export const TYPOGRAPHY_CSS_PATH = 'node_modules/@sbb-esta/angular-public/typography.css';
+
+/** Name of the Angular module that enables Angular browser animations. */
+export const BROWSER_ANIMATIONS_MODULE_NAME = 'BrowserAnimationsModule';
+
+/** Name of the module that switches Angular animations to a noop implementation. */
+export const NOOP_ANIMATIONS_MODULE_NAME = 'NoopAnimationsModule';
+
+export default function(options: Schema): Rule {
+  return chain([addAnimationsModule(options), addTypographyToAngularJson(options)]);
+}
+
+function addTypographyToAngularJson(options: Schema): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    addTypographyToStylesNode('build', options, host, context);
+    addTypographyToStylesNode('test', options, host, context);
+    return host;
+  };
+}
+
+function addTypographyToStylesNode(
+  buildOrTest: 'build' | 'test',
+  options: Schema,
+  tree: Tree,
+  context: SchematicContext
+): Tree {
+  const workspace = getWorkspace(tree);
+  const projectName = options.name || workspace.defaultProject;
+
+  if (
+    !projectName ||
+    !workspace ||
+    !workspace.projects ||
+    !workspace.projects[projectName] ||
+    !workspace.projects[projectName].architect ||
+    !workspace.projects[projectName].architect![buildOrTest] ||
+    !workspace.projects[projectName].architect![buildOrTest]!.options
+  ) {
+    context.logger.error('Cannot add typography. Please add it manually to angular.json');
+    return tree;
+  }
+
+  if (!(workspace.projects[projectName].architect![buildOrTest]!.options! as any).styles) {
```

Comment:
Maybe assign options to a variable, so you don't have to use the full path every time